src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -378,6 +378,40 @@ JNI_METHOD(void, establishPaseConnectionByAddress)
     }
 }
 
+JNI_METHOD(void, pauseCommissioning)
+(JNIEnv * env, jobject self, jlong handle)
+{
+    ChipLogProgress(Controller, "pauseCommissioning() called");
+    chip::DeviceLayer::StackLock lock;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    wrapper->Controller()->PauseCommissioning();
+}
+
+JNI_METHOD(void, resumeCommissioning)
+(JNIEnv * env, jobject self, jlong handle)
+{
+    ChipLogProgress(Controller, "resumeCommissioning() called");
+    chip::DeviceLayer::StackLock lock;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    wrapper->Controller()->ResumeCommissioning();
+}
+
+JNI_METHOD(void, updateCommissioningNetworkCredentials)
+(JNIEnv * env, jobject self, jlong handle, jobject networkCredentials)
+{
+    ChipLogProgress(Controller, "updateCommissioningNetworkCredentials() called");
+    chip::DeviceLayer::StackLock lock;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    wrapper->Controller()->ResumeCommissioning();
+
+    CommissioningParameters commissioningParams = CommissioningParameters();
+    wrapper->ApplyNetworkCredentials(commissioningParams, networkCredentials);
+    wrapper->UpdateNetworkCredentials(commissioningParams);
```

Comment:
Might want to handle any returned errors here? If they cannot be handled here, perhaps you'd want to return them to the caller of this JNI_METHOD?

Reply:
modified to throw an exception